lib/go/thrift/header_protocol.go

```diff
@@ -303,3 +303,17 @@ func (p *THeaderProtocol) ReadBinary() (value []byte, err error) {
 func (p *THeaderProtocol) Skip(fieldType TType) error {
 	return p.protocol.Skip(fieldType)
 }
+
+// GetResponseHeadersFromClient is a helper function to get the read THeaderMap
+// from the last response received from the given client.
+//
+// If the last response was not sent over THeader protocol,
+// an empty map will be returned.
```

Comment:
Should be `nil will be returned` since nil maps have slightly different semantics. We could return an initialized empty map instead, but that seems like an unnecessary allocation.

Reply:
Changed to `a nil map will be returned`.